### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by falzm**
This change introduces a dedicated `Dockerfile` in order to avoid specifying build details (that are duplicating the ones in the `Makefile`) into the `Jenkinsfile`. Since I'm not familiar at all with the Jenkinsfile format, can you please tell me if I'm on the right track?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM registry.internal.exoscale.ch/exoscale/golang:1.10 as build
+
+RUN mkdir -p /go/src/github.com/exoscale/zlocker
+ADD . /go/src/github.com/exoscale/zlocker
+WORKDIR /go/src/github.com/exoscale/zlocker
+
+RUN make
```
Comment (greut) [inline Dockerfile:7]: arrrgh :wink:

File: Jenkinsfile [hunk 2]
```diff
@@ -70,11 +70,5 @@ def golint() {
...
 
 def build() {
-  docker.withRegistry('https://registry.internal.exoscale.ch') {
-    def image = docker.image('registry.internal.exoscale.ch/exoscale/golang:1.10')
-    image.inside("-u root --net=host -v ${env.WORKSPACE}/src:/go/src/github.com/exoscale/zlocker") {
-      sh 'cd /go/src/github.com/exoscale/zlocker && dep ensure'
-      sh 'cd /go/src/github.com/exoscale/zlocker && CGO_ENABLED=0 go build -ldflags "-s"'
-    }
-  }
+  docker.build('registry.internal.exoscale.ch/exoscale/zlocker:latest').withRun('--net=host')
```
Comment (greut) [inline Jenkinsfile]: https://github.com/exoscale/exoip/blob/master/Jenkinsfile#L84-L109

**Comment by greut** [comment]
@falzm whatever works. As we move away from dep to go modules, my favorite direction would be to also drop the Makefile (hence the over dramatic _arrrgh_)

File: Jenkinsfile (hunk 2 above)
Comment (greut) [reply Jenkinsfile]: iirc you have to have the two steps in order to produce the local binary for the .deb and the container image.

**Comment by falzm** [comment]
> @falzm whatever works. As we move away from dep to go modules, my favorite direction would be to also drop the Makefile (hence the over dramatic _arrrgh_)
> 


Sure, I'd rather do it with a Go module but in this case I wanted to do as little change as possible in the original design in order to save time.

**Comment by greut** [comment]
:tada: !

File: Jenkinsfile [hunk 3]
```diff
@@ -10,29 +12,37 @@ node {
...
-            build()
+          "docker build": {
+            image = docker(repo)
+          },
+          "deb package": {
+            build(repo)
+            gitPbuilder('bionic')
           }
         )
       }
```
Comment (greut) [inline Jenkinsfile:33]: so the parallel stages could be:
- lint + build
- docker + deb
- registry + aptly

Comment (falzm) [reply Jenkinsfile:33]: Yes but if that's OK with you I have lost enough time on this trivial change for the moment, could this be addressed later?

Comment (greut) [reply Jenkinsfile:33]: On the other hand, you've learned a ton.

**Review by greut — APPROVED** [review]
:lock:

File: Jenkinsfile (hunk 3 above)
Comment (falzm) [reply Jenkinsfile:33]: Agreed ;)

**Comment by falzm** [comment]
@brutasse any opinion on this PR?

**Review by brutasse — APPROVED** [review]
@falzm go for it :shipit:

**Comment by falzm** [comment]
🎉

**Pull request merged**